<launch>

    <arg name="gui" default="0" />
    <arg name="paused" default="0" />

    <rosparam command="load" ns="tfd_modules" file="$(find tfd_modules)/config/timeout_90s_5min_nontemporal.yaml" />

    <!-- start executive -->
    <node pkg="continual_planning_executive" type="continual_planning_executive" args="" respawn="false" name="continual_planning_executive" output="screen" >
        <param name="start_paused" value="$(arg paused)" />
        <param name="domain_file" value="$(find planner_benchmarks)/modular/tidyup/domain_modules.pddl" />

        <param name="locations" value="$(find tidyup_actions)/launch/locations.dat" />

        <!-- tolerance to use when determining if navigation has reached a target (i.e. we are AT a target)
            if nav_target_tolerance_relative_to_move_base is true, nav_base_local_planner_ns should give
            the namespace that holds the xy_goal_tolerance parameter the base_local_planner uses.
        -->
        <param name="nav_target_tolerance_xy" value="0.25" />
        <param name="nav_target_tolerance_yaw" value="0.25" />
        <param name="nav_target_tolerance_relative_to_move_base" value="false" />
        <param name="nav_base_local_planner_ns" value="/move_base/TrajectoryPlannerROS" />

        <param name="door_location_file" value="$(find tidyup_actions)/config/doors.dat" />

        <rosparam command="load" file="$(find tidyup_actions)/config/tidyup_actions.yaml" />
        <rosparam command="load" file="$(find tfd_modules)/config/tfdm_interface.yaml" />
        <!-- navstack_init "fixed_frame" "goal_tolerance used in planner requests" 
             "goal_tolerance is relative to move_base's xy_goal_tolerance either
             0 for false or the namespace to base_local_planner params" 
             1 means - guess namespace -->
        <rosparam param="planner_options">
            - (planning_scene_navstack_init@libplanner_modules_pr2.so /map 0.05 1)
            - (putdown_init@libputdown_modules.so /map)
        </rosparam>
        <env name="LD_LIBRARY_PATH" value="$(find planner_modules_pr2)/lib:$(optenv LD_LIBRARY_PATH)" />
    </node>

    <node if="$(arg gui)" pkg="continual_planning_monitor" type="continual_planning_monitor" name="continual_planning_monitor" />

</launch>

